edx_rest_framework_extensions/tests/test_permissions.py

```diff
@@ -51,19 +51,53 @@ def test_has_permission(self, is_staff):
 class IsUserInUrlTests(TestCase):
     """ Tests for the IsUserInUrl permission class. """
 
-    @ddt.data(True, False)
-    def test_has_permission(self, has_user_in_url):
-        user = factories.UserFactory(username='foo')
+    def _create_request(self, user, username_in_param=None, username_in_resource=None):
         url = '/'
-        if has_user_in_url:
-            url += '?username={}'.format(user.username)
+        if username_in_param:
+            url += '?username={}'.format(username_in_param)
         request = RequestFactory().get(url)
         request.user = user
+        if username_in_resource:
+            request.parser_context = dict(kwargs=dict(username=username_in_resource))
+        return request
+
+    @ddt.data(True, False)
+    def test_user_in_url_param(self, has_user_in_url):
+        user = factories.UserFactory(username='this_user')
+        request = self._create_request(
+            user,
+            username_in_param=user.username if has_user_in_url else 'another_user',
+        )
         self.assertEqual(permissions.IsUserInUrl().has_permission(request, None), has_user_in_url)
 
+    @ddt.data(True, False)
+    def test_user_in_url_resource(self, has_user_in_resource):
+        user = factories.UserFactory(username='this_user')
+        request = self._create_request(
+            user,
+            username_in_resource=user.username if has_user_in_resource else 'another_user',
+        )
+        self.assertEqual(permissions.IsUserInUrl().has_permission(request, None), has_user_in_resource)
+
+    def test_resource_takes_precedence_over_param(self):
+        user = factories.UserFactory(username='this_user')
+        request = self._create_request(
+            user,
+            username_in_resource='another_user',
```

Comment:
Nit: Could use ddt with username_in_resource, username_in_param, and expected_has_permission, but fine as-is as well.